### PR TITLE
add func MkdirAll

### PR DESCRIPTION
**Pull request by lostz**
add func MkdirAll

### CODE REVIEW

**Comment by mdlayher** [comment]
I like the idea, but this needs at least a test or two to verify its functionality.

File: client.go [hunk 1]
```diff
@@ -694,6 +694,37 @@ func (c *Client) Mkdir(path string) error {
 	}
 }
 
+func (c *Client) MkdirAll(path string) error {
```
Comment (mdlayher) [inline client.go]: Needs a documentation comment.

File: client.go [hunk 2]
```diff
@@ -694,6 +694,37 @@ func (c *Client) Mkdir(path string) error {
...
+			return err
+		}
+	}
+	err = c.Mkdir(path)
+	if err != nil {
+		return err
+	}
+
+	return nil
+
```
Comment (mdlayher) [inline client.go]: Remove this blank line.

File: client.go [hunk 3]
```diff
@@ -694,6 +694,37 @@ func (c *Client) Mkdir(path string) error {
...
+		j--
+	}
+	if j > 1 {
+		// Create parent
+		err = c.MkdirAll(path[0 : j-1])
+		if err != nil {
+			return err
+		}
+	}
+	err = c.Mkdir(path)
```
Comment (davecheney) [inline client.go]: return c.Mkdir(path)

**Comment by davecheney** [comment]
Why don't we just make c.Mkdir do this automatically ? Then we don't have to have two functions with subtly different semantics.

**Comment by mdlayher** [comment]
I'd be okay with that as well. While I feel there could be value in matching the API of the `os` package, I would prefer recursive directory creation 9 times out of 10 in practice.

**Comment by mordyovits** [comment]
Please match the os API.  Anything else would catch people (me) by surprise.

**Comment by davecheney** [comment]
This PR appears to be stale. Closing

If someone wants to reactivate it later please _raise an issue first_.

**Pull request closed without merging**